ansible/roles/controller/tasks/deploy.yml

```diff
@@ -236,6 +236,11 @@
 
       "CONFIG_whisk_transactions_header": "{{ transactions.header }}"
 
+- name: merge extra env variables
+  set_fact:
+    controller_env: "{{ controller_env | default({}) | combine(controller.extraEnv) }}"
+  when: controller.extraEnv | default(false)
```

Comment:
Why is the `when` needed in the first place? Shouldn't we be able to specify: `extraEnv: "{{ controller_extraEnv | default({}) }}"` and run the combine step no matter what?

Reply:
I assumed we need it the extra safety, your proposing just dropping the `when` line?
 since in group_vars we already default to `default({})` and extra check is not need it.

Reply:
I see define the default and then drop the `when` I will give this a test and update